ninja.lua

```diff
@@ -245,6 +249,17 @@ local function getcxxflags(toolset, cfg, filecfg)
 	return buildopt .. cppflags .. cxxflags .. defines .. includes .. forceincludes
 end
 
+local function getmasmflags(toolset, cfg, filecfg)
+	local defines = ninja.list(table.join(toolset.getdefines(filecfg.defines), toolset.getundefines(filecfg.undefines)))
+
+	local extra = ""
+	if filecfg.exceptionhandling == "SEH" then
+		extra = extra .. " /safeseh"
```

Comment:
That is msc specific :-/
Should ideally be in premake-core.
here, at least protect with toolset check.
See if it a workaround can be done in "Some toolset fixes/helper" section.